### ISSUE TITLE
Getting Issue With Nirmala UI Font [ Character Issue ]

### ISSUE THREAD

**Issue by DevMensIT** [libgd/libgd#861]
**Describe the bug**

Getting Face this kind of issue when i put text inside image using gd library the output is not getting valid but when i echo that text i'm getting proper output. 

Wrote this  : "આત્મા" output showing me this "આતમા" 
Wrote this : "હર્ષ" output showing me this "હરષ"

is there any solution regarding this ?

Note : Text from Windows Language [ Indian ] which is installed on Windows Computer. Supported By Microsoft.

**Comment by vapier** [comment]
please attach an example program showing the problem, and the build config of your gd lib, and the version of libraries you're linking against

**Comment by TheDevManIT** [comment]
Both is in php : 


The true output I want is : 
![image](https://user-images.githubusercontent.com/11617013/220894987-bb28dab9-c378-4e7d-a6fc-0515fd10222c.png)


But The Output I'm getting is here : 
![image](https://user-images.githubusercontent.com/11617013/220894698-4f60f086-7ced-4433-ba3e-92a0fe325222.png)


Basically its grammatical mistake I'm getting here.

**Comment by cmb69** [comment]
> Both is in php :

It's often a good idea to report such issues at https://github.com/php/php-src/issues, and to provide an [SSCCE](https://sscce.org/). If the PHP developers refer you to libgd, you can still open an issue here.

> Wrote this : "હર્ષ" output showing me this "હરષ"

That may be a general issue with RTL (right to left) text, or possibly caused by libraqm (I don't think that PHP reports whether libraqm is used by libgd). It may also be an issue with combining characters.

If this issue is still unresolved, I'll try to have a closer look.

**Comment by DevMensIT** [comment]
Yes. Please look into it because i tried to change font it same issue and if i do bold it replace with square like not supporting

**Comment by cmb69** [comment]
Okay, since I don't know this script, first step is to check what we have: https://3v4l.org/U7FQP reveals that there are four code points (`GUJARATI LETTER HA` (`U+0AB9`), `GUJARATI LETTER RA` (`U+0AB0`), `GUJARATI SIGN VIRAMA` (`U+0ACD`), `GUJARATI LETTER SSA` (`U+0AB7`)). Gujarati is a LTR script, so RTL issues can be excluded.

Then I run the following script (PHP's bundled libgd, since I'm assuming you're running on Windows):
````php
$im = imagecreatetruecolor(64, 32);
imagefilledrectangle($im, 0, 0, imagesx($im) - 1, imagesy($im) - 1, 0x000000);
imagettftext($im, 16, 0, 10, 20, 0xffffff, "Nirmala", "હર્ષ");
imagegif($im);
````
and I get:
![gd861](https://github.com/user-attachments/assets/14042d9b-b157-4cc0-8472-a63ecc5a4586)

So, yeah, an issue with combining characters, basically the same as #758. However, in the meantime libgd has optional support for complex text layout (libraqm), so when trying that with libgd 2.3.3 and libraqm, I get the desired result:
![gd861](https://github.com/user-attachments/assets/8411c3d2-3124-4cc8-a028-1a551e7641a2)

So the good news is that this is not a libgd issue; the bad news is that when you use PHP's bundled libgd (only option on Windows), then you're screwed. See https://github.com/php/php-src/issues/17352.
